pkg/repository/repository.go

```diff
@@ -38,13 +38,31 @@ type repositoryInterface interface {
 	getSHA() (string, error)
 }
 
-// NewRepository creates a repository controller
-func NewRepository(path string) Repository {
-	repoUrl, err := giturls.Parse(path)
+type repositoryURL struct {
+	url.URL
+}
+
+// String is a custom implementation for the url where User is removed
+func (r repositoryURL) String() string {
+	r.URL.User = nil
```

Comment:
it is safe to do it? if we remove the user from the repo could not be affected future uses about the repository object ? could we work with a copy for this case ?

Reply:
makes sense, updating the PR

Reply:
@AdrianPedriza i've added a test for this function